public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -123,21 +133,106 @@
+                }
+            };
+
+
 
+
+
+    /**
+     * Converter function from backend representation (used with Hibernate) to
+     * client representation (generated by Swagger).
+     */
+    private static final Function<AchillesResult, org.pmiops.workbench.model.AchillesResult>
+            TO_CLIENT_ACHILLES_RESULT =
+            new Function<AchillesResult, org.pmiops.workbench.model.AchillesResult>() {
+                @Override
+                public org.pmiops.workbench.model.AchillesResult apply(AchillesResult o) {
+
+                    return new org.pmiops.workbench.model.AchillesResult()
+                            .id(o.getId())
+                            .analysisId(o.getAnalysisId())
+                            .stratum1(o.getStratum1())
+                            .stratum2(o.getStratum2())
+                            .stratum3(o.getStratum3())
+                            .stratum4(o.getStratum4())
+                            .stratum5(o.getStratum5())
+                            .stratum5Name(o.getStratum5Name())
+                            .countValue(o.getCountValue());
                 }
             };
 
 
+    /* Final api functions put on top */
+    @Override
+    public ResponseEntity<DbDomainListResponse> getDomainFilters() {
+        List<DbDomain> domains=dbDomainDao.findByDbType("domain_filter");
+        DbDomainListResponse resp=new DbDomainListResponse();
+        resp.setItems(domains.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+    }
+
+    @Override
+    public ResponseEntity<DbDomainListResponse> getSurveyList() {
+        List<DbDomain> domains=dbDomainDao.findByDbTypeAndAndConceptIdNotNull("survey");
+        DbDomainListResponse resp=new DbDomainListResponse();
+        resp.setItems(domains.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+    }
+
+
+
+    @Override
+    public ResponseEntity<QuestionConceptListResponse> getSurveyResults(String surveyConceptId) {
+        /* Set up the age and gender names */
+        // Too slow and concept names wrong so we hardcode list
+        // List<Concept> genders = conceptDao.findByConceptClassId("Gender");
+
+        long longSurveyConceptId = Long.parseLong(surveyConceptId);
+
+        // Get questions for survey
+        List<QuestionConcept> questions = questionConceptDao.findSurveyQuestions(longSurveyConceptId);
+
+        // Get survey definition
+        QuestionConceptListResponse resp = new QuestionConceptListResponse();
+        DbDomain survey = dbDomainDao.findByConceptId(longSurveyConceptId);
+        resp.setSurvey(TO_CLIENT_DBDOMAIN.apply(survey));
+
+        // Get all analyses for question list and put the analyses on the question objects
+        if (!questions.isEmpty()) {
+            // Put ids in array for query to get all results at once
+            List<String> qlist= new ArrayList();
```

Comment:
If you prefer, you can do questions.stream().map(QuestionConcept::getConceptId()).collect(Collectors.toList()) and skip the loop.

Reply:
Thanks. I tried a second. Got red and couldn't figure it out quick. I have to convert conceptId to string too. I'll just leave it for now.